libvast/src/system/index.cpp

```diff
@@ -585,6 +590,8 @@ index(index_actor::stateful_pointer<index_state> self,
       }
       // Get all potentially matching partitions.
       auto candidates = self->state.meta_idx.lookup(expr);
+      for (const auto& [id, _] : self->state.unpersisted)
```

Comment:
Is this a gcc/clang extension that you don't need `[[maybe_unused]]` here?

Reply:
Good question, I tried a bit locally and I couldn't get gcc/clang to emit a warning at all as long as part of a structured binding is used:

```
benno@poincare:~/code/cpp$ cat unused2.cpp 
#include <iostream>
#include <utility>
#include <vector>


int main() {
  for (auto& [x, _] : std::vector<std::pair<int,int>>{std::make_pair(3,4)})
	 std::cout << x << std::endl;
}
benno@poincare:~/code/cpp$ clang++ -std=c++17 unused2.cpp -Wall -Wextra
benno@poincare:~/code/cpp$ 
```

So I think the other instances of this in the codebase might just be unnecessary.